src/jit/lower.cpp

```diff
@@ -628,20 +620,14 @@ GenTree* Lowering::LowerSwitch(GenTree* node)
 
                 // Now, build the conditional statement for the current case that is
                 // being evaluated:
-                // GT_JTRUE
-                //   |__ GT_COND
-                //          |____GT_EQ
-                //                 |____ (switchIndex) (The temp variable)
-                //                 |____ (ICon)        (The actual case constant)
-                GenTreePtr gtCaseCond =
-                    comp->gtNewOperNode(GT_EQ, TYP_INT, comp->gtNewLclvNode(tempLclNum, tempLclType),
-                                        comp->gtNewIconNode(i, TYP_INT));
-                /* Increment the lvRefCnt and lvRefCntWtd for temp */
+                // GT_CMP switchValue, caseValue
+                // GT_JCC(EQ)
```

Comment:
> What kind of diffs are you getting out of our test code?

I only ran jitdiff on fx, I've yet to run it on tests:
* the `TEST` to `BT` transform described above saves 332 bytes (most in `System.Private.Uri` which happens to heavily use a 64-bit "flags" enum). No regressions.
* `(x >> y) & 1` appears to be a dud, 0 differences. I need to try the other variant of this pattern - `x & (1 << y)`.
* `BT` based switch table saves 3781 bytes. Diffs look like (example from https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/BlobUtilities.cs#L337):
```diff
        41FFC9               dec      r9d
        4183F92C             cmp      r9d, 44
        7722                 ja       SHORT G_M55886_IG08
-       418BC1               mov      eax, r9d
-       4C8D0D22000000       lea      r9, [reloc @RWD00]
-       458B0C81             mov      r9d, dword ptr [r9+4*rax]
-       4C8D15BFFFFFFF       lea      r10, G_M55886_IG02
-       4D03CA               add      r9, r10
-       41FFE1               jmp      r9
+       B8FFF0FF7F           mov      eax, 0x7FFFF0FF
+       440FA3C8             bt       eax, r9d
+       7306                 jae      SHORT G_M55886_IG08
```
This seems by far the most useful use of BT and there are probably more opportunities for it:
* Roslyn sometimes use binary search instead of `switch` and my current implementation doesn't recognize that. It should be able to recognize code like `if (x == 2 || x == 3 || x == 8)` as well.
* This may be combined with switch table to binary search conversion (which the JIT currently doesn't do). That's quite a bit of work and outside the scope of this experiment.